pkg/util/ready/ready.go

```diff
@@ -0,0 +1,65 @@
+package ready
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// SimpleHTTPClient to aid in mocking
+type SimpleHTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// URL returns boolean if url is ready
+func URL(cli SimpleHTTPClient, url string) (bool, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return false, err
+	}
+	var resp *http.Response
+	resp, err = cli.Do(req)
+	if err != nil {
+		return false, err
+	}
+	return resp.StatusCode == http.StatusOK, nil
+}
+
+// URLPoolState returns when URL pool propagates to  expected state or an error after 20 min
+func URLPoolState(ctx context.Context, log *logrus.Entry, cli SimpleHTTPClient, pool []string, expectedState bool) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 20*time.Minute)
+	defer cancel()
```

Comment:
it is better for the caller to set the context - trying to move to that pattern generally